salt/modules/match.py

```diff
@@ -348,6 +351,16 @@ def filter_by(lookup,
     for key in lookup:
         params = (key, minion_id) if minion_id else (key, )
         if expr_funcs[tgt_type](*params):
+            if merge:
+                if not isinstance(merge, collections.Mapping):
+                    raise SaltException(
+                        'filter_by merge argument must be a dictionary.')
+
+                if lookup[key] is None:
+                    return merge
+                else:
+                    salt.utils.dictupdate.update(lookup[key], copy.deepcopy(merge))
```

Comment:
It would be nice to also be able to set merge_list=True on this.

Reply:
Added the `merge_lists=True` option to the dictupdate.update call.

Reply:
No, i think it should be configurable, it shouldn't always be on by default.

Reply:
Updated `merge_lists` to be a configurable option in `match.filter_by`.

Reply:
Awesome, thanks!